tripal/src/Hook/TripalHooks.php

```diff
@@ -0,0 +1,392 @@
+    $config = \Drupal::config('tripal_admin.settings');
+    // Add some variables for all javasript to use for building URLs.
+    $clean_urls = $config->get('clean_url', 0);
+    $tripal_path = \Drupal::service('extension.list.module')->getPath('tripal');
+    // Add a JS library.
+    $variables['#attached']['library'][] = 'tripal/tripal-js';
+    $variables['#attached']['drupalSettings']['tripal']['tripalJS']['baseurl'] = $base_url;
+    $variables['#attached']['drupalSettings']['tripal']['tripalJS']['isClean'] = $clean_urls;
+    $variables['#attached']['drupalSettings']['tripal']['tripalJS']['tripal_path'] = $tripal_path;
+    // Make sure the date time settings are the way Tripal will insert them,
+    // otherwise PostgreSQL version that may have a different datestyle setting
+    // will fail when inserting or updating a date column in a table.
+    // @todo we do this for every page? Maybe move to rebuild().
+    // @see Issue #2360.
+    \Drupal::database()->query("SET DATESTYLE TO :style", [':style' => 'MDY']);
+
+    // Ask users to do the registration form.
+    // @upgrade when Issue #45 is closed.
+    // if (\Drupal::currentUser()->hasPermission('administer tripal')) {
+    // if (empty($config->get('tripal_site_registration') ?: FALSE)
+    // || !($config->get('disable_tripal_reporting') ?: FALSE)) {
+    // \Drupal::messenger()->addWarning(t('Please register your Tripal Site.
+    // Registering provides important information that will help secure funding
+    // for continued improvements to Tripal.
+    // <a href="admin/tripal/register">Click to register now or opt out</a>.'));
+    // }
+    // }
+  }
+
+  /**
+   * Implements hook_form_alter for the field_config_edit_form.
+   *
+   * Adds the form elements to the field settings form for setting controlled
+   * vocabulary. The function only does this for fields that are not
+   * Tripal fields but which are attached to a tripal content type.
+   */
+  #[Hook('form_alter')]
+  public function formFieldConfigEditFormAlter(&$form, FormStateInterface $form_state, $form_id) {
+    if ($form_id === 'field_config_edit_form') {
+      $form_object = $form_state->getFormObject();
+      if (method_exists($form_object, 'getEntity')) {
+        /** @var \Drupal\field\Entity\FieldConfig $field **/
+        $field = $form_object->getEntity();
+        if (method_exists($field, 'getTargetEntityTypeId')) {
+          if ($field->getTargetEntityTypeId() == 'tripal_entity') {
+            $elements = TripalFieldItemBase::buildFieldTermForm($field, $form, $form_state);
+            $form['settings'] = $form['settings'] + $elements;
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Implements hook_preprocess_field().
+   *
+   * This is to permit HTML markup in the page title field, for
+   * example on an organism page, to display genus and species
+   * in italics.
+   */
+  #[Hook('preprocess_field')]
+  public function preprocessField(&$variables) {
```

Comment:
This hook is borderline to include in the `TripalTheme` render class; however, it is not restricted to render-specific changes. I think it might be best to move this into a new class `TripalEntityHooks` which can include all entity, entity storage and field hooks. I actually have at least one more hook to go in this class in PR #2281.